__tests__/parser-test.js

```diff
@@ -0,0 +1,88 @@
+const {describe, expect, test, beforeAll, it } = require('@jest/globals');
+const { parseForESLint }  = require('../src/parser/gjs-gts-parser');
+const { traverse }  = require( '../src/parser/transforms');
+const { SourceCode } = require( 'eslint');
+
+describe('transform', () => {
+  let text, result;
+  beforeAll(() => {
+    text = `
+    import { ExternalLink, Link, service } from 'limber-ui';
+
+const ReportIssue = <template>
+  If the tutorial navigated you here,
+  <ExternalLink href="https://github.com/NullVoxPopuli/limber/issues">please report the issue</ExternalLink>.
+  ❤️
+</template>;
+
+const CurrentPath = <template>
+  {{#let (service "docs") as |docs|}}
+    <code>{{docs.currentPath}}</code>
+  {{/let}}
+</template>;
+
+const BackToStart = <template>
+  You may also try going
+  <Link href="/" style="width: max-content; display: inline-block;">back to the beginning</Link>
+</template>;
+
+export const NotFound = <template>
+  Prose for the current tutorial, <CurrentPath />, could not be found.
+
+  Please check the URL and try again,
+  or navigate to a different tutorial chapter.
+
+  <br /><br />
+  <ReportIssue />
+
+  <br /><br /><br>
+  <BackToStart />
+</template>;
+    `;
+
+    result = parseForESLint(text, {
+      filePath: 'example.gts',
+      comment: true,
+      loc: true,
+      range: true,
+      tokens: true
+    });
+  })
+
+  it('all tokens are correct', () => {
+    expect(result.ast.tokens.map(t => ({
+      value: t.value,
+      loc: t.loc,
+      range: t.range
+    }))).toMatchSnapshot();
+  })
+
+  it('all nodes have tokens', () => {
+    const source = new SourceCode({ ...result, text });
+    traverse(result.visitorKeys, result.ast, (path) => {
+      expect(source.getTokens(path.node)).not.toHaveLength(0);
+    });
+  });
+
+  it('node ranges are correct', () => {
+    const nodes = [];
+    traverse(result.visitorKeys, result.ast, (path) => {
+      nodes.push(path.node);
+    });
+    expect(nodes.map(n => [n.type, text.slice(...n.range)])).toMatchSnapshot();
+  });
+
+  it('token ranges are correct', () => {
+    const tokens = result.ast.tokens;
+    expect(tokens.map(n => [n.type, text.slice(...n.range)])).toMatchSnapshot();
```

Comment:
Why snapshots instead of inline snapshots?

Reply:
doesnt support prettier v3... :(

Reply:
Works fine in vitest